centreon/src/Core/ServiceSeverity/Application/UseCase/UpdateServiceSeverity/UpdateServiceSeverity.php

```diff
@@ -0,0 +1,175 @@
+                    )
+            ) {
+                $this->error(
+                    'Service severity not found',
+                    ['severity_id' => $severityId]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Service severity'));
+
+                return;
+            }
+
+            if (null === ($severity = $this->readServiceSeverityRepository->findById($severityId))) {
+                $this->error(
+                    'Service severity not found',
+                    ['severity_id' => $severityId]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Service severity'));
+
+                return;
+            }
+
+            $this->validateNameOrFail($dto->name, $severity);
+            $this->validateIconOrFail($dto->iconId);
+
+            $severity->setName($dto->name);
+            $severity->setAlias($dto->alias);
+            $severity->setLevel($dto->level);
+            $severity->setIconId($dto->iconId);
+            $severity->setActivated($dto->isActivated);
+
+            $this->writeServiceSeverityRepository->update($severity);
+
+            $presenter->setResponseStatus(new NoContentResponse());
+        } catch (ServiceSeverityException $ex) {
+            $presenter->setResponseStatus(
+                match ($ex->getCode()) {
+                    ServiceSeverityException::CODE_CONFLICT => new ConflictResponse($ex),
+                    default => new ErrorResponse($ex),
+                }
+            );
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (AssertionFailedException $ex) {
+            $presenter->setResponseStatus(new InvalidArgumentResponse($ex));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(
+                new ErrorResponse(ServiceSeverityException::editServiceSeverity($ex))
+            );
+            $this->error((string) $ex);
+        }
+    }
+
+    /**
+     * @param string $name
+     * @param ServiceSeverity $severity
+     *
+     * @throws ServiceSeverityException
+     */
+    private function validateNameOrFail(string $name, ServiceSeverity $severity): void
+    {
+        if (
+            $name !== $severity->getName()
+            && $this->readServiceSeverityRepository->existsByName(new TrimmedString($name))
+        ) {
+            $this->error(
+                'Service severity name already exists',
+                ['severity_name' => $name]
+            );
+
+            throw ServiceSeverityException::serviceNameAlreadyExists();
+        }
+    }
+
+    /**
+     * @param int $iconId
+     *
+     * @throws ServiceSeverityException
+     */
+    private function validateIconOrFail(int $iconId): void
+    {
+        if (0 === $iconId || ! $this->readViewImgRepository->existsOne($iconId)) {
```

Comment:
The usecase is not supposed to know how IDs are managed by the repository.
```suggestion
        if (! $this->readViewImgRepository->existsOne($iconId)) {
```